src/Core/System.php

```diff
@@ -72,6 +72,7 @@ public static function callstack($depth = 4)
 				}
 			} elseif (!in_array($func['function'], $ignore)) {
 				$callstack[] = $func['function'];
+				$func['class'] = '';
```

Comment:
Why this?

Reply:
Because of this: https://github.com/friendica/friendica/blob/bd1056a61e49b7ec2073d2fba5dc45f23188b2f0/src/Core/System.php#L68
`$previous['class']` is evaluated but not checked for existence like `$func['class']` is.

Reply:
Okay. But would be some "default" function better here? ```$func['class']``` often will have a value.

Reply:
No, this is the case where we execute a global function (like the ones in`boot.php`).

Reply:
The backtrace array only has the `class` key when the call concerns a class method, but the key is absent for global function calls. We do check for the `class` key on a new item in the call stack, but then we stored it as is in `$previous` where we didn't check the existence of `class`, hence the need for manual initialization when it doesn't exist.